scylla/src/transport/session_test.rs

```diff
@@ -1345,4 +1345,49 @@ async fn test_schema_types_in_metadata() {
             frozen: true
         }
     );
-}
\ No newline at end of file
+}
+
+#[tokio::test]
+async fn test_column_kinds_in_metadata() {
+    let uri = std::env::var("SCYLLA_URI").unwrap_or_else(|_| "127.0.0.1:9042".to_string());
+    let session = SessionBuilder::new().known_node(uri).build().await.unwrap();
+
+    session
+        .query("DROP KEYSPACE IF EXISTS test_metadata_ks", &[])
+        .await
+        .unwrap();
+
+    session
+        .query("CREATE KEYSPACE test_metadata_ks WITH REPLICATION = {'class' : 'SimpleStrategy', 'replication_factor' : 1}", &[])
+        .await
+        .unwrap();
+
+    session.query("USE test_metadata_ks", &[]).await.unwrap();
+
+    session
+        .query(
+            "CREATE TABLE IF NOT EXISTS t (
+                    a int,
+                    b int,
+                    c int,
+                    d int STATIC,
+                    e int,
+                    f int,
+                    PRIMARY KEY ((c, e), b, a)
+                  )",
+            &[],
+        )
+        .await
+        .unwrap();
+
```

Comment:
To be 100% sure that the test works with clusters with size larger than one node, you should wait for schema agreement here.